### ISSUE TITLE
Add integer support (Lua 5.3)

### ISSUE THREAD

**Issue by scoder** [scoder/lupa#146]
Lua 5.3 has [introduced](https://www.lua.org/manual/5.3/readme.html#changes) support for integer values, in addition to the previous C double floating point values. Lupa should support these and map Python `int` objects to Lua integers, it possible. It seems ok to drop support for Lua 5.1/2 for this feature, then releasing as Lupa 2.0.

- add tests
- add the new type to `py_from_lua()` and `py_to_lua()`.
- find a reasonable fallback for large integers (preferably configurable as `LuaRuntime` parameter), either rejecting them or converting them to a Lua float.

**Comment by mfrigerio17** [comment]
Is this solved in Lupa 1.10?
I still get a floating point number in Lua when using Python's `len` builtin:

```python
import lupa
lua = lupa.LuaRuntime()
f = lua.execute("return function(seq) return print( python.builtins.len(seq) ), print(4) end")
f( ['a', 'b', 'c', 'd'] )
```
This prints:
```
4.0
4
```
I expect the integer `4` also for the first print.

Python 3.9.5
Lua 5.3.3
Lupa 1.10

If I _return_ the length instead of printing it (within Lua), the return value in Python is actually an integer:

```python
f = lua.execute("return function(seq) return python.builtins.len(seq) end")
type( f( ['a', 'b', 'c', 'd'] ) )
```
Prints: `int`

**Comment by scoder** [comment]
Yes, it's in 1.10. Can't say right now what goes wrong here. Could you investigate a bit deeper?

**Comment by Satellite-Galaxy** [comment]
I am not sure this was fixed I am using 1.13 and am having the same issue.
